src/btree/row_srch.c

```diff
@@ -389,9 +390,9 @@ __wt_row_search(WT_CURSOR_BTREE *cbt, WT_ITEM *srch_key, bool insert, WT_REF *le
      * In some cases we expect we're comparing more than a few keys with matching prefixes, so it's
      * faster to avoid the memory fetches by skipping over those prefixes. That's done by tracking
      * the length of the prefix match for the lowest and highest keys we compare as we descend the
-     * tree. The high boundary is reset on each new page, the lower boundary is maintained.
```

Comment:
Perhaps we could modify the comment as 
`The high boundary is reset on each new page, and the lower boundary is maintained unless the page index changes before reaching the leaf page.`

Reply:
I also modify only to reset the high boundary if there is a race